REGRESSION(289493@main): Pinch zooming PDFs in Safari is disorienting

aprotyas opened the pull request:
#### 806258dc2944ef654ab2e1a1e179a11cf20f1c6c
<pre>
REGRESSION(289493@main): Pinch zooming PDFs in Safari is disorienting
<a href="https://bugs.webkit.org/show_bug.cgi?id=286737">https://bugs.webkit.org/show_bug.cgi?id=286737</a>
<a href="https://rdar.apple.com/143857710">rdar://143857710</a>

Reviewed by Wenson Hsieh.

289493@main makes ENABLE_LEGACY_PDFKIT_PLUGIN==0 on macOS 15.2+. So,
with these changes, those platforms unconditionally report false for
WebPage::handlesPageScaleGesture(). This leads to ViewGestureController
calling adjustTransientZoom() on a page&apos;s drawing area, rather than
calling scalePage() on the page directly, during magnification,
resulting in the reported symptom.

We fix this by simply guarding the plugin check inside
WebPage::handlesPageScaleGesture() behind ENABLE_PDF_PLUGIN rather than
ENABLE_LEGACY_PDFKIT_PLUGIN.

* Source/WebKit/WebProcess/Plugins/PluginView.cpp:
(WebKit::PluginView::platformLayer const):

Drive-by cleanup: There is no need to compile this out for non-legacy
plugins.

* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::handlesPageScaleGesture):

Canonical link: <a href="https://commits.webkit.org/289577@main">https://commits.webkit.org/289577@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2ab94538438dec1b0f918473a65458bb85e0e381

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/87331 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/6841 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/41693 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/92193 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/38071 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/7242 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/15025 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/67483 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/25214 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/90333 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/5532 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/79036 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/47811 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/5323 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/37188 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/75746 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/34292 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/94080 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/14493 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/10665 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/76296 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/14771 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/74892 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/75504 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/18583 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/19902 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/18292 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/7426 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/14512 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/19805 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/14257 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/17700 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/16038 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

whsieh reviewed (approved):
r=me. Test failures seem unrelated.

aprotyas committed:
REGRESSION(289493@main): Pinch zooming PDFs in Safari is disorienting
https://bugs.webkit.org/show_bug.cgi?id=286737
rdar://143857710

Reviewed by Wenson Hsieh.

289493@main makes ENABLE_LEGACY_PDFKIT_PLUGIN==0 on macOS 15.2+. So,
with these changes, those platforms unconditionally report false for
WebPage::handlesPageScaleGesture(). This leads to ViewGestureController
calling adjustTransientZoom() on a page's drawing area, rather than
calling scalePage() on the page directly, during magnification,
resulting in the reported symptom.

We fix this by simply guarding the plugin check inside
WebPage::handlesPageScaleGesture() behind ENABLE_PDF_PLUGIN rather than
ENABLE_LEGACY_PDFKIT_PLUGIN.

* Source/WebKit/WebProcess/Plugins/PluginView.cpp:
(WebKit::PluginView::platformLayer const):

Drive-by cleanup: There is no need to compile this out for non-legacy
plugins.

* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::handlesPageScaleGesture):

Canonical link: https://commits.webkit.org/289577@main
